Minio/DataModel/Select/SelectResponseStream.cs

```diff
@@ -0,0 +1,239 @@
+                System.Buffer.BlockCopy(headers,0,inputArray,prelude.Length+ preludeCRC.Length, headerLength);
+                System.Buffer.BlockCopy(payload,0,inputArray,prelude.Length+ preludeCRC.Length+headerLength, payloadLength);
+                System.Buffer.BlockCopy(messageCRC,0,inputArray, totalLength - 4,messageCRC.Length);
+
+                // write real data to inputArray
+                Crc32Algorithm.ComputeAndWriteToEnd(inputArray); // last 4 bytes contains CRC
+                // transferring data or writing reading, and checking as final operation
+                if (!Crc32Algorithm.IsValidWithCrcAtEnd(inputArray))
+                {
+                    throw new ArgumentException("Prelude CRC Mismatch");
+                }
+
+                Dictionary<String, String> headerMap = extractHeaders(headers);
+
+                string value = null;
+                if (headerMap.TryGetValue(":message-type", out value))
+                {
+                    if (value.Equals(":error"))
+                    {
+                        string errorCode = null;
+                        string errorMessage = null;
+                        headerMap.TryGetValue(":error-code", out errorCode);
+                        headerMap.TryGetValue(":error-message", out errorMessage);
+                        throw new MinioException(errorCode + ":" + errorMessage);
+                    }
+                }
+                if (headerMap.TryGetValue(":event-type", out value))
+                {
+                    if (value.Equals("End"))
+                    {
+                        // throw new UnexpectedShortReadException("Insufficient data");
+                        this._isProcessing = false;
+                        break;
+                    }
+                    if (value.Equals("Cont") || payloadLength < 1)
+                    {
+                        continue;
+                    }
+                    if (value.Equals("Progress"))
+                    {
+                        ProgressMessage progress = new ProgressMessage();
+                        using (var stream = new MemoryStream(payload))
+                        progress = (ProgressMessage)new XmlSerializer(typeof(ProgressMessage)).Deserialize(stream);
+                        this.Progress = progress;
+                    }
+                    if (value.Equals("Stats"))
+                    {
+                        Console.WriteLine("payload|"+Encoding.UTF7.GetString(payload));
+                        StatsMessage stats = new StatsMessage();
+                        using (var stream = new MemoryStream(payload))
+                        stats = (StatsMessage)new XmlSerializer(typeof(StatsMessage)).Deserialize(stream);
+                        this.Stats = stats;
+                    }
+                    if (value.Equals("Records"))
+                    {
+                        this.Payload.Write(payload,0, payloadLength);
```

Comment:
If i understand correctly, `Payload` is the response writer that the caller/user will be listening to? @poornas

Reply:
that's correct.